### PR TITLE
feat: update Category page with feature toggle

### PR DESCRIPTION
**Pull request by Aylie-Chou**
- [TWREPORTER-381](https://twreporter-org.atlassian.net/browse/TWREPORTER-381)
- please test it with twreporter-npm-packages [#303](https://github.com/twreporter/twreporter-npm-packages/pull/303)

### CODE REVIEW

File: src/containers/Category.js [hunk 1]
```diff
@@ -209,30 +250,53 @@ function postsProp(state, listId, page) {
...
-  const pathSegment = _.get(props, 'match.params.category')
-  const catId = categoryConst.ids[pathSegment]
-  const catLabel = categoryConst.labels[pathSegment]
-  const pathname = _.get(location, 'pathname', `/categories/${pathSegment}`)
+  const categoryPath = _.get(props, 'match.params.category')
+  const subcategoryPath = _.get(props, 'match.params.subcategory')
+  const category = CATEGORY[categoryPath]
+  const subcategory = SUBCATEGORY[subcategoryPath]
 
+  let listId = `${category && category.id}`
```
Comment (duidae) [inline src/containers/Category.js:258]: how about simply `category?.id`

File: src/data-loaders/category-list-page.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ export default function loadData({ location, match, store }) {
   }
 
   const pathSegment = _.get(match, 'params.category', '')
-  const catId = categoryConst.ids[pathSegment]
+  const catId = CATEGORY[pathSegment] && CATEGORY[pathSegment].id
```
Comment (duidae) [inline src/data-loaders/category-list-page.js:37]: how about simply `CATEGORY[pathSegment]?.id`

**Comment by duidae** [comment]
Mostly looking good & it's a massive work! Just a minor issue about feature toggle, I ran into some errors when `make lint`, will we also update `package.json` in this PR?

File: src/containers/Category.js (hunk 1 above)
Comment (Aylie-Chou) [reply src/containers/Category.js:259]: bable 6 do not deal with optional chaining, I'll create another PR to upgrade babel then fix this 😂

**Comment by Aylie-Chou** [comment]
updated. ([9d48e90](https://github.com/twreporter/twreporter-react/pull/2228/commits/9d48e90d37a7301a0be43bfdebb7f6360f3fa06f))

- update @twreporter/core version to include feature toggle

**Review by duidae — APPROVED** [review]
LGTM

**Pull request merged**